extensions/sql-format/src/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { Clipboard } from '@raycast/api';
+import { getPreferenceValues, showHUD, showToast, Toast } from '@raycast/api';
+
+import { format } from 'sql-formatter';
+
+export default async () => {
+  const output = formatSQL((await Clipboard.readText()) || '');
+  if (output) {
+    await copyFormattedSQL(output);
+  }
+};
+
+// format sql
+export function formatSQL(text: string) {
+  const trimmedText = text.trim();
+  let output = '';
+  try {
+    if (!validateSQL(trimmedText)) {
+      showToastMessage();
+      return;
+    }
+
+    const tabWidth = getTabWidth();
+    const keywordCase = getKeywordCase();
+    const options = {
+      tabWidth: tabWidth === 'tab' ? 8 : parseInt(tabWidth),
+      keywordCase: keywordCase,
+    };
+
+    output = format(trimmedText, options);
+  } catch (error) {
+    showToastMessage();
+    return;
+  }
+  return output;
+}
+
+// copy or auto paste output
+export async function copyFormattedSQL(result: string) {
+  if (autoPasteEnabled()) {
+    await Clipboard.paste(result);
+    await showHUD('✅ Pasted succesfully!');
+  } else {
+    await Clipboard.copy(result);
+    await showHUD('✅ Copied succesfully!');
+  }
+}
+
+// user preferences settings
+interface Preferences {
+  tabWidth: tabWidthType;
+  keywordCase: keywordCaseType;
+  autopaste: boolean;
+}
+
+type tabWidthType = '2' | '4' | 'tab';
+type keywordCaseType = 'preserve' | 'upper' | 'lower';
```

Comment:
(non-blocking) These are not necessary, as _Preferences_ is present in _raycast-env.d.ts_. See https://github.com/raycast/extensions/pull/14155 for more details.

```suggestion
```

Reply:
I try it works, thank you tell for me the tricks. I really appreciate your code review of my code, it made it better.

Reply:
No worries, one additional thing I would suggest is to add another command to "Format Selected SQL". You can find that I am adding similar command in https://github.com/raycast/extensions/pull/14155. This saves an extra step (not copying to clipboard), if someone just wants to format a selected text.

Reply:
I looked at the PR cited in the comment above, and it just so happened that I had the same idea as you, and I plan to add it in the next update. Thanks for your suggestions, I think getting from the clipboard and selecting text for formatting should be two very common operations.